src/Doctrine/EntityRegenerator.php

```diff
@@ -24,39 +23,31 @@
  */
 final class EntityRegenerator
 {
-    private $doctrineRegistry;
+    private $doctrineHelper;
     private $fileManager;
     private $generator;
     private $projectDirectory;
     private $overwrite;
-    private $metadataFactory;
 
-    public function __construct(ManagerRegistry $doctrineRegistry, FileManager $fileManager, Generator $generator, string $projectDirectory, bool $overwrite)
+    public function __construct(DoctrineHelper $doctrineHelper, FileManager $fileManager, Generator $generator, string $projectDirectory, bool $overwrite)
     {
-        $this->doctrineRegistry = $doctrineRegistry;
+        $this->doctrineHelper = $doctrineHelper;
         $this->fileManager = $fileManager;
         $this->generator = $generator;
         $this->projectDirectory = $projectDirectory;
         $this->overwrite = $overwrite;
-        $this->metadataFactory = new DoctrineMetadataFactory($this->doctrineRegistry);
```

Comment:
The `DoctrineMetadataFactory` calls should be deleted entirely now, correct?

Reply:
yes, and DoctrineMetadataFactory class too I think